### DISCUSSION TITLE
Memory leak?

### DISCUSSION THREAD

**Discussion by milo-ucla** [Pryaxis/TShock#1701] [General]
I have noticed that after starting TShock.exe multiple times (with the quit command) that my computer begins to lag after ~5 restarts. Is it possible that the program is not deallocating space correctly after it closes down?

**Comment by hakusaro** [comment]
It's possible that the program isn't fully exiting? Can you open task manager or equivalent on Linux/macOS and screenshot the details/processing tab, descending by memory use, after doing this?

**Comment by moisterrific** [comment]
From my experience of starting and shutting down a test server countless number of times, I’ve not seen any slowdown to my computer‘s performance. For reference my CPU is a quad core i7-4870 @2.5 GHz and I have 16 GB DDR4 RAM. Perhaps your PC does not meet the recommended system requirements?

**Comment by tylerjwatson** [comment]
Hi,

As we've got no details on your situation, from a technical standpoint memory leaks (at least in the traditional sense) are not possible after a process has exited because the kernel has to keep track of memory it gives you so it can release those regions when the process exits.  What a kernel _doesn't_ do, is nudge a running process and say "hey, you lost this".  Memory leaks happen because the _process_ has lost track of the memory given to it and either has to continue to ask for more again or crash.

TerrariaServer _will_, however, cause a lot of stuff to swap in and out of virtual memory on disk on Windows, particularly if you rent a server with a relatively small amount of RAM.  Slowdowns will become very apparent very quickly if the running instance of TerrariaServer consumes almost all of the working set of memory, forcing other stuff out.
